### ISSUE TITLE
Audit logs on older ubuntu hosts

### ISSUE THREAD

**Issue by tailhook** [tailhook/vagga#71]
The problem similar to #52 is actually also with other commands, for example `su`. This is related to ubuntu audit log. Pam error message:
```
<82>Oct  5 15:11:00 su[3]: PAM audit_log_acct_message() failed: Operation not permitted
<83>Oct  5 15:11:00 su[3]: pam_authenticate: System error
```
And su error looks simply like:
```
su: System error
```

Some manual: http://linux.die.net/man/3/audit_log_acct_message
Similar issue in docker: https://github.com/docker/docker/issues/5899

You may avoid the issue by adding "audit=0" to the kernel command line (in grub config). Still long-term solution must be implemented.

**Comment by popravich** [comment]
Note: *This (`su`) works fine in Ubuntu 15.04*

**Comment by tailhook** [comment]
Workaround for docker: https://github.com/sequenceiq/docker-pam

**Comment by tailhook** [comment]
So rebuilding pam with disabled audit is okay (https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile). But I believe it might be slow to do on each container build :(

**Comment by tailhook** [comment]
I'm going to close this. It's only old ubuntu (there is a rumor that fedora and centos too) bug. There are couple of things to avoid that:

1. Use `sudo` instead of `su` (alternatively [`gosu`](https://github.com/tianon/gosu) like [docker does](https://github.com/docker-library/postgres/blob/cd294bf8dfdf4a74b2077aa6413fa579f9bf07de/9.4/Dockerfile) )
2. Use [pam with disabled audit](https://github.com/sequenceiq/docker-pam/blob/master/ubuntu-14.04/Dockerfile)
3. Set `audit=0` in linux kernel boot options (e.g. in grub)

Note the first two options are done inside the container, and last one must be done on host system.
Some newer kernel might probably fix this too (but I've not tested, 3.19 has bug AFAIK).

For now I'm closing this as WONTFIX.

We may introduce a `!Sh`-like command that does setuid on its own. Or do some other fix for this bug (not sure what we can do? replace pam in ubuntu by default?). So if you want to hack on this, Pull Requests are welcome.

**Comment by tailhook** [comment]
Okay, another good solution is to use `chpst` from `runit` package instead of `sudo`:
```
- !Install [runit]
- !Sh "chpst -u username command"
```
